libs/ui/calendar/brain/src/lib/brn-calendar.directive.ts

```diff
@@ -0,0 +1,206 @@
+import { BooleanInput, NumberInput } from '@angular/cdk/coercion';
+import {
+	ChangeDetectorRef,
+	Directive,
+	Injector,
+	afterNextRender,
+	booleanAttribute,
+	computed,
+	contentChild,
+	contentChildren,
+	inject,
+	input,
+	model,
+	numberAttribute,
+	signal,
+} from '@angular/core';
+import { injectDateAdapter } from '@spartan-ng/ui-date-time';
+import { BrnCalendarCellButtonDirective } from './brn-calendar-cell-button.directive';
+import { BrnCalendarHeaderDirective } from './brn-calendar-header.directive';
+import { provideBrnCalendar } from './brn-calendar.token';
+
+@Directive({
+	selector: '[brnCalendar]',
+	standalone: true,
+	providers: [provideBrnCalendar(BrnCalendarDirective)],
+})
+export class BrnCalendarDirective<T> {
+	/** Access the date adapter */
+	protected readonly dateAdapter = injectDateAdapter<T>();
+
+	/** Access the change detector */
+	private readonly changeDetector = inject(ChangeDetectorRef);
+
+	/** Access the injector */
+	private readonly injector = inject(Injector);
+
+	/** The minimum date that can be selected.*/
+	public readonly min = input<T>();
+
+	/* * The maximum date that can be selected. */
+	public readonly max = input<T>();
+
+	/** Determine if the date picker is disabled. */
+	public readonly disabled = input<boolean, BooleanInput>(false, {
+		transform: booleanAttribute,
+	});
+
+	/** The selected value. */
+	public readonly date = model<T>();
+
+	/** Whether a specific date is disabled. */
+	public readonly dateDisabled = input<(date: T) => boolean>(() => false);
+
+	/** The day the week starts on */
+	public readonly weekStartsOn = input<Weekday, NumberInput>(0, {
+		transform: (v: unknown) => numberAttribute(v) as Weekday,
+	});
+
+	/** The default focused date. */
+	public readonly defaultFocusedDate = input<T>();
+
+	/** @internal Access the header */
+	protected readonly header = contentChild(BrnCalendarHeaderDirective);
```

Comment:
```suggestion
	public readonly header = contentChild(BrnCalendarHeaderDirective);
```

We need this because we access this variable in [brn-calendar-grid.directive.ts](https://github.com/spartan-ng/spartan/pull/480/files#diff-583e25e213df9cf5c9db39f56be34a967d0879838a7cafae7fe22b12047fa5ad)